src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -639,7 +639,7 @@ N*P).
 template <DGLDeviceType XPU, typename IdType>
 CSRMatrix COOToCSR(COOMatrix coo) {
   if (!coo.row_sorted) {
-    const int64_t num_threads = omp_get_num_threads();
+    const int64_t num_threads = omp_get_max_threads();
```

Comment:
I'm more curious why/when `omp_get_num_threads` would return 1.

Reply:
It always return 1 when it is not run in parallel region. 
Acording to the 5.0 documentation:
"Effect: The omp_get_num_threads routine returns the number of threads in the team that is executing the parallel region to which the routine region binds. If called from the sequential part of a program, this routine returns 1."

For _omp_get_max_threads_:
"Effect: The value returned by omp_get_max_threads is the value of the first element of the nthreads-var ICV of the current task. **This value is also an upper bound on the number of threads that could be used to form a new team if a parallel region without a num_threads clause were encountered after execution returns from this routine**."